tests/network/vmi_multus.go

```diff
@@ -1024,6 +1024,21 @@ var _ = Describe("[Serial]SRIOV", func() {
 					"SR-IOV VF is expected to exist in the guest after migration")
 			})
 
+			It("should re-attach sriov devices on source VMI when failed during setup", func() {
+				By("annotate the VMI with functest force migration failure annotation")
+				vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+				vmi.Annotations = map[string]string{v1.FuncTestForceLauncherMigrationFailureAnnotation: ""}
+				vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Update(vmi)
+				Expect(err).ToNot(HaveOccurred())
+
+				migration := tests.NewRandomMigration(vmi.Name, vmi.Namespace)
+				migrationUID := tests.RunMigrationAndExpectFailure(virtClient, migration, tests.MigrationWaitTime)
+				tests.ConfirmVMIPostMigrationFailed(virtClient, vmi, migrationUID)
```

Comment:
Isn't this covered by `tests.RunMigrationAndExpectFailure` which was prev called?

Reply:
`RunMigrationAndExpectFailure` wait until migration phase is failure 
while `ConfirmVMIPostMigrationFailed` checks that the VMI status.MigrationState reflects migration failure

Reply:
Ohh... one looks at the `VMIM` object while the other on the `VMI`, right?